src/alchemlyb/estimators/ti_.py

```diff
@@ -166,3 +177,129 @@ def separate_dhdl(self):
                 new.attrs = self.dhdl.attrs
                 dhdl_list.append(new)
         return dhdl_list
+
+    def gaussian_quadrature(self, dHdl, means, variances):
+        """
+        numerically estimate the free energy using gaussian quadrature with 
+        lambda values suggested in Amber manual
+
+        Parameters
+        ----------
+        dHdl : DataFrame
+        means : DataFrame
+        variances: DataFrame
+        """
+        # extract the lambda values used in the simulations
+        lambdas = means.reset_index()[means.index.names[:]].iloc[:].values.sum(axis=1)
+        num_lambdas = len(lambdas)
+        # check if the lambda values used match the common ones suggested in Amber manual and assign cooresponding weights
+        if num_lambdas==1:
+            try:
+                reference_lambdas = np.round([0.5], 4)
+                assert (lambdas == reference_lambdas).all()
```

Comment:
assert is not the correct way to check input. If there's anything wrong with user data you need to raise an exception, probably a ValueError.

Furthermore, *never* use `==` to check floating point number equality. Instead use `np.allclose()` or similar.